oarepo_ui/resources/config.py

```diff
@@ -170,18 +173,53 @@ def search_app_config(self, identity, api_config, overrides={}, **kwargs):
         opts.update(kwargs)
         return SearchAppConfig.generate(opts, **overrides)
 
-    @property
-    def custom_fields(self):
-        # TODO: currently used by forms only, implement custom fields loading
-        return {
-            "ui": {},
+    def custom_fields(self, **kwargs):
+        api_service = current_service_registry.get(self.api_service)
+        # get the record class
+        record_class = getattr(api_service, "record_cls", None) or getattr(
+            api_service, "draft_cls", None
+        )
+        ui = []
+        ret = {
+            "ui": ui,
         }
+        if not record_class:
+            return ret
+        # try to get custom fields from the record
+        for fld_name, fld in sorted(inspect.getmembers(record_class)):
+            if isinstance(fld, InlinedCustomFields):
+                prefix = ""
+            elif isinstance(fld, CustomFields):
+                prefix = fld.key + "."
+            else:
+                continue
+
+            ui_config = self._get_custom_fields_ui_config(fld.config_key, **kwargs)
+            if not ui_config:
+                continue
+
+            for section in ui_config:
+                ui.append(
+                    {
+                        **section,
+                        "fields": [
+                            {
+                                **field,
+                                "field": prefix + field["field"],
+                            }
+                            for field in section.get("fields", [])
+                        ],
+                    }
+                )
+        return ret
```

Comment:
Possible bug, is updated `ui` also reflected by `ret` value here?

Reply:
@mirekys Hey, the function does work, ui is a reference to an array and later the for loop is pushing to this array.

Reply:
Are you sure? :) There is a pending Sonar issue about this function always returning the same value no matter what happens in the for loop

Reply:
@mirekys Zdar, I can tell you for sure that it works. However, it was written by Miro. I think sonarlint is not maybe smart enough to get the code in this instance. Because ret is set to  ret = {
            "ui": ui,
        }
        
        And ui is an array, to which the for loop later appends stuff.
        
        So it will return either ui:[], or ur :[itemsfromthefoorloop]